### PR TITLE
feat: Implement `DeepEvalEvaluator`

### PR DESCRIPTION
**Pull request by shadeMe**
Related to #250.

We introduce `DeepEvalEvaluator`, a component that uses the [DeepEval LLM evaluation framework](https://docs.confident-ai.com/docs/getting-started) to calculate evaluation metrics for RAG pipelines (among others). Refer https://github.com/deepset-ai/haystack/pull/6784 for an overview of the API design.

This PR introduces the following user-facing classes:
- `DeepEvalMetric` - A enumeration that lists the supported DeepEval metrics. Currently, only those metrics that are related to RAG pipelines are supported.
- `DeepEvalEvaluator` - Th pipeline component interfaces with the evaluation framework. It accepts a single metric and its optional parameters. The inputs to the pipeline are dynamically configured depending on the metric. This is done with help of a metric descriptor table that contains metadata concerning input/output conversion formats, expected inputs/outputs, etc.

The output of the component is a nested list of metric results. Each input can have one or more results, depending on the metric. Each result is a dictionary containing the following keys and values:

- `name` - The name of the metric.
- `score` - The score of the metric.
- `explanation` - An optional explanation of the score.

### CODE REVIEW

File: integrations/deepeval/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+git_describe_command = 'git describe --tags --match="integrations/deepeval-v[0-9]*"'
+
+[tool.hatch.envs.default]
+dependencies = ["coverage[toml]>=6.5", "pytest"]
+[tool.hatch.envs.default.scripts]
+test = "pytest {args:tests}"
+test-cov = "coverage run -m pytest {args:tests}"
+cov-report = ["- coverage combine", "coverage report"]
+cov = ["test-cov", "cov-report"]
+
```
Comment (masci) [inline integrations/deepeval/pyproject.toml:49]: The `readme_sync` CI job will fail when we merge this on `main`. Should we add at least a `hatch run docs` no-op command in this PR?

**Review by masci — COMMENTED** [review]
I'm not familiar with deepeval but code looks good to me. I left a question about api docs now that we merged the CI job that will attempt to run `hatch run docs` on integrations' changes merged on `main`. Another option might be making that CI job more resilient.

File: integrations/deepeval/pyproject.toml (hunk 1 above)
Comment (shadeMe) [reply integrations/deepeval/pyproject.toml:49]: Good point, I'll add that.

**Review by masci — APPROVED** [review]
Thanks for adding the docs, LGTM

**Pull request merged**